api/v1beta1/cluster_types.go

```diff
@@ -1037,7 +1045,13 @@ func (c *Cluster) GetClassKey() types.NamespacedName {
 	if c.Spec.Topology == nil {
 		return types.NamespacedName{}
 	}
-	return types.NamespacedName{Namespace: c.GetNamespace(), Name: c.Spec.Topology.Class}
+
+	namespace := c.Spec.Topology.ClassNamespace
+	if c.Spec.Topology == nil || c.Spec.Topology.ClassNamespace == "" {
```

Comment:
```suggestion
	if namespace == "" {
```
Above we are returning when c.Spec.Topology == nil

Reply:
Resolving as this code is gone, new code is fine